### PR TITLE
Support parsing MBIDs for roles.

### PR DESCRIPTION
**Pull request by kgarner7**
Currently a draft PR. Focusing on just the standard individual tags (rather than performer). Working: vorbis/m4a (Picard)
Not working: tak (not parsed at all), wma (order is wrong)

### CODE REVIEW

File: adapters/taglib/taglib.go [hunk 1]
```diff
@@ -122,16 +122,16 @@ func parseTIPL(tags map[string][]string) {
...
-			tags[role] = append(tags[currentRole], strings.Join(currentValue, " "))
+			tags[role] = append(tags[role], strings.Join(currentValue, " "))
 		}
 	}
 
 	var currentRole string
 	var currentValue []string
 	for _, part := range strings.Split(tipl[0], " ") {
-		if _, ok := tiplMapping[part]; ok {
+		if _, ok := tiplMapping[strings.ToLower(part)]; ok {
```
Comment (deluan) [inline adapters/taglib/taglib.go:132]: Can't we just set the correct dj-mix case in the `tiplMapping` map, and avoid the ToLower here? This logic is too fragile and is based on exact matches, making it case insensitive could increase the chances of false positives.

Ideally we should parse TIPL directly, and not the string representation, but this will have to wait.

Comment (kgarner7) [reply adapters/taglib/taglib.go:132]: Picard specifically does DJ-mix. I have no idea what other tools use.

File: adapters/taglib/taglib.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ func parseTIPL(tags map[string][]string) {
 	addRole := func(currentRole string, currentValue []string) {
 		if currentRole != "" && len(currentValue) > 0 {
 			role := tiplMapping[currentRole]
-			tags[role] = append(tags[currentRole], strings.Join(currentValue, " "))
+			tags[role] = append(tags[role], strings.Join(currentValue, " "))
```
Comment (deluan) [inline adapters/taglib/taglib.go:125]: Nice catch! To cover this situation, please add another DJ-mix to the test so we are sure it is appending to the right slice

Comment (kgarner7) [reply adapters/taglib/taglib.go:125]: So, 
```go
tags["tipl"] = []string{"arranger Andrew Powell DJ-mix François Kevorkian DJ-mix Jane Doe engineer Chris Blair"}
```
?

**Comment by kgarner7** [comment]
All mbids are now parsed. I'd like to add some more edge case testing (e.g., what happens if you have less mbids than artists, more, no mbids), but the core should be there. End to end tests for the seven formats specified work

File: model/metadata/map_participants_test.go [hunk 3]
```diff
@@ -518,4 +518,76 @@ var _ = Describe("Participants", func() {
...
+
+				Expect(roles[0].Name).To(Equal("a"))
+				Expect(roles[1].Name).To(Equal("b"))
+
+				Expect(roles[0].MbzArtistID).To(Equal("f634bf6d-d66a-425d-888a-28ad39392759"))
+				Expect(roles[1].MbzArtistID).To(Equal("3dfa3c70-d7d3-4b97-b953-c298dd305e12"))
+			}
+		})
+
+		It("should refuse duplicate names if no mbid specified", func() {
```
Comment (deluan) [inline model/metadata/map_participants_test.go:576]: What is the expected behaviour if there are duplicates AND mbids specified?

**Review by deluan — COMMENTED** [review]
Looks good. I added a test case and some refactoring.

File: model/metadata/map_participants_test.go (hunk 3 above)
Comment (kgarner7) [reply model/metadata/map_participants_test.go:576]: It's not possible currently, because the ID is by artist name
